### PR TITLE
chore: Added extra analytics events for license

### PR DESCRIPTION
**Pull request by vishnu-gp**
## Description

Added extra events for license-related analytics events. Also did some functional breakdown of code

## Type of change

- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?

- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/AnalyticsServiceCEImpl.java [hunk 1]
```diff
@@ -273,16 +273,24 @@ public <T extends BaseDomain> Mono<T> sendObjectEvent(AnalyticsEvents event, T o
...
+        final String eventTag = isNonResourceEvent ? event.getEventName() : event.getEventName() + "_" + object.getClass().getSimpleName().toUpperCase();
+
+        return eventTag;
+    }
+
+    /**
+     * To get non resource events list
+     * @return List of AnanlyticsEvents
+     */
+    public List<AnalyticsEvents> getNonResourceEvents() {
```
Comment (vivonk) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/AnalyticsServiceCEImpl.java:287]: Shall we define such specific events in  `AnalyticsEvents.java` as a static final list?

**Review by vivonk — COMMENTED** [review]
Added a very small comment, else LGTM

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/AnalyticsServiceCEImpl.java (hunk 1 above)
Comment (vishnu-gp) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/AnalyticsServiceCEImpl.java:287]: The plan is to override this method from EE to add EE-specific events. Hence added like this

**Review by vivonk — APPROVED** [review]

**Pull request merged**